src/appchooserdialog.c

```diff
@@ -132,6 +132,12 @@ show_more (AppChooserDialog *dialog)
   gtk_window_set_resizable (GTK_WINDOW (dialog), TRUE);
   gtk_widget_hide (dialog->more_row);
 
+  for (i = 0; i < INITIAL_LIST_SIZE - 1; i++)
```

Comment:
Use `g_strv_length()` instead.

Reply:
Much better. Not sure what I was thinking. :S

Reply:
Actually I will use an assert() here instead. It's guaranteed that we always have more choices than INITIAL_LIST_SIZE if we reach this function.

Reply:
A `g_warning()` or a `g_return_val_if_fail()` would be better, the app calling the portal isn't responsible for bugs in the code, and there are other apps that might be using the portal...

Reply:
Well this function will only ever be called when there are more apps available than fit in the initial list. So an assert is appropriate here: it will only be hit if there is a bug in xdg-desktop-chooser-gtk itself.